include/lsst/ctrl/events/Host.h

```diff
@@ -0,0 +1,74 @@
+// -*- lsst-c++ -*-
+
+/* 
+ * LSST Data Management System
+ * Copyright 2008-2014  AURA/LSST.
+ * 
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the LSST License Statement and 
+ * the GNU General Public License along with this program.  If not, 
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/** \file Host.h
+  *
+  * \ingroup events
+  *
+  * \brief defines the Host class
+  *
+  */
+
+#ifndef LSST_CTRL_EVENTS_HOST_H
+#define LSST_CTRL_EVENTS_HOST_H
+
+#include <stdlib.h>
+#include <iostream>
+#include <vector>
+#include <list>
+#include <boost/shared_ptr.hpp>
+
+using namespace std;
+
+namespace lsst {
+namespace ctrl {
+namespace events {
+
+/**
+ * @brief This object represents the host that is being used to transmit events,
+ *        giving quick access to often-used host information.
+ */
+class Host {
+public:
+    Host() {}
+
+    ~Host() {}
+
+    static Host& getHost();
+
+    unsigned int getIPAddress();
+    std::string getHostName();
+
+protected:
+    static Host* thisHost;
```

Comment:
Then this object should probably be renamed to represent the originating host. The intent was to get this information once and reuse it, since it is used so frequently.